### ISSUE TITLE
OpenResty and Nginx CVE-2019-20372

### ISSUE THREAD

**Issue by keinproblem** [openresty/openresty#577]
Dear OpenResty Community,

I recently came across following CVE: https://nvd.nist.gov/vuln/detail/CVE-2019-20372
As elaborated, it is affecting versions of nginx prior to 1.17.7.

I got to know that "OpenResty® is not an Nginx fork. It is a higher level application and gateway platform using nginx as a component. " https://openresty.org/en/

Anyways it seems like OpenResty still somehow uses the actual nginx releases for its build: https://travis-ci.org/openresty/docker-openresty/jobs/612375286#L5152

The whole versioning thing is quite confusing for me, personally.

I created this issue in order to get some clearance regarding the CVE mentioned above.
Is OpenResty affected by the CVE-2019-20372 found in Nginx?

If this is not the case, then how would be the systematic approach to identify and correlate nginx vulnerability finding with OpenResty? 
In the first place it looks like the version strategy of OpenResty is adapting the versioning of latest nginx releases - assumingly, OpenResty 1.15.8 has an nginx 1.15.8 running, as seen in the travis-ci build job posted above.

Do nginx vulnerabilities generally apply to OpenResty?

**Comment by agentzh** [comment]
@keinproblem Yes, OpenResty includes Nginx as a component and does include its own nginx core patches. We're about to make a new OpenResty release based on the latest nginx mainline core. cc @thibaultcha

**Comment by keinproblem** [comment]
Thanks @agentzh, that addresses my major concern! 
So considering CVE-2019-20372, OpenResty would fix this vulnerability in one of the more recent releases, which do base on the one of the latest, non-vulnerable, nginx mainline core?

I am excited to see the new OpenResty releases in the upcoming future.

**Comment by agentzh** [comment]
OpenResty 1.17.8.1 RC1 will be out in the next few days. @thibaultcha is helping with the release engineering work.

**Comment by keinproblem** [comment]
I have seen that RC1 is out now: https://github.com/openresty/lua-resty-core/issues/286#issuecomment-587304264
and that a compatibility matrix was created for the lua-nginx-module https://github.com/openresty/lua-nginx-module#nginx-compatibility

Are there already and bigger conclusions that could lead to a ETA for the sable release?

I found this page with respect to upgrading OpenResty: https://openresty.org/en/upgrading.html
Would it still be valid for the current case?

**Comment by rhuddleston** [comment]
Any ETA on rc1 or even a GA release of 1.17.8.1?
